packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,513 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoint.dart';
+import 'slot_layout.dart';
+import 'slot_layout_config.dart';
+
+/// A parent Widget takes in multiple [SlotLayout] components and places them
+/// into their appropriate positions on the screen.
+
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.topNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.horizontalBody = true,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the screen.
```

Comment:
You might explain which side is the "beginning" side in RTL and LTR. I mean, it's obvious in retrospect, but maybe not obvious that you're calling it the "beginning" side because of directionality.